### PR TITLE
Updated examples to use  instead of using urlRenderOverride

### PR DESCRIPTION
**Pull request by pythonissam**

### CODE REVIEW

**Comment by pythonissam** [comment]
@snoyberg I feel like current implementation of `urlParamRenderOverride` doesn't fit with the description. Is this okay?

File: book/asciidoc/yesod-typeclass.asciidoc [hunk 1]
```diff
@@ -430,17 +430,18 @@ static files from a separate domain]. The advantage to this approach is that
...
 For example, the scaffolding defines this method as:
 
 
 [source, haskell]
 ----
-urlRenderOverride y (StaticR s) =
-    Just $ uncurry (joinPath y (Settings.staticRoot $ settings y)) $ renderRoute s
-
-urlRenderOverride _ _ = Nothing
+    urlParamRenderOverride :: site
```
Comment (snoyberg) [inline book/asciidoc/yesod-typeclass.asciidoc]: This isn't actually defined in the scaffolding. I'd rather update the text to say something like "you could use the following to provide special rendering of static routes" and then keep the example as it was previously.

Comment (pythonissam) [reply book/asciidoc/yesod-typeclass.asciidoc]: Got it. I'll do so:)

**Comment by pythonissam** [comment]
@snoyberg How is this?

**Comment by pythonissam** [comment]
done

File: book/asciidoc/yesod-typeclass.asciidoc [hunk 2]
```diff
@@ -430,24 +430,18 @@ static files from a separate domain]. The advantage to this approach is that
...
+                           -> Route site
+                           -> [(T.Text, T.Text)] -- ^ query string
+                           -> Maybe Builder
+    urlParamRenderOverride _ _ _ = Nothing
```
Comment (snoyberg) [inline book/asciidoc/yesod-typeclass.asciidoc]: I'm sorry, I must not be finding the right words. I like the previous example demonstrating `joinPath` and `renderRoute`, and I'd like to keep it.

**Comment by pythonissam** [comment]
@snoyberg Would you review it?

**Comment by snoyberg** [comment]
I've touched it up a bit and merged to master.

**Pull request merged**